blinky/source/coordinator/coordinator.c

```diff
@@ -1,48 +1,25 @@
 #include "coordinator.h"
 
-#include <stddef.h>
-
 #include "indication.h"
 #include "movement.h"
 #include "timer.h"
 
-timer timerIndicationOn = NULL;
-enum mode_enum { MODE_IDLE, MODE_DETECTED } mode = MODE_IDLE;
+timer timerIndicationOn;
 indication indicator;
 
-void coordinator_init(uint8_t base_ms, indication indication_instance) {
-    if (timerIndicationOn == NULL) {
-        timerIndicationOn = eiger_timer_create(base_ms);
-    }
+void coordinator_init(timer t, indication indication_instance) {
+    timerIndicationOn = t;
     indicator = indication_instance;
     indication_reset(indicator);
-    mode = MODE_IDLE;
 }
 
-void coordinator_destroy() { eiger_timer_destroy(timerIndicationOn); }
-
-void set_mode() {
+void coordinator_run() {
     if (movement_has_rotated()) {
-        mode = MODE_DETECTED;
-    }
-    if (eiger_timer_elapsed_time_ms(timerIndicationOn) > 2000) {
-        mode = MODE_IDLE;
-    }
-}
-
-void action() {
-    if (mode == MODE_IDLE) {
-        indication_reset(indicator);
-        eiger_timer_reset(timerIndicationOn);
-    }
-    if (mode == MODE_DETECTED) {
         indication_indicate(indicator);
         eiger_timer_update(timerIndicationOn);
         movement_reset();
+    } else if (eiger_timer_elapsed_time_ms(timerIndicationOn) > 2000) {
```

Comment:
This state will never be reached because the timer will only be updated once. The function movement_reset will set movement has rotated to false and then eiger_timer_update function will not be executed.